src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -117,8 +117,13 @@ private function addService(Definition $definition, ?string $id, \DOMElement $pa
             $service->setAttribute('lazy', 'true');
         }
         if (null !== $decorated = $definition->getDecoratedService()) {
-            list($decorated, $renamedId, $priority) = $decorated;
+            list($decorated, $renamedId, $priority, $onDecoratedInvalid) = $decorated;
             $service->setAttribute('decorates', $decorated);
+
+            if (\in_array($onDecoratedInvalid, [ContainerInterface::NULL_ON_INVALID_REFERENCE, ContainerInterface::IGNORE_ON_INVALID_REFERENCE])) {
+                $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE === $onDecoratedInvalid ? 'nulled' : 'ignored';
```

Comment:
"null" / "ignore" ("nulled" / "ignored" wouldn't be consistent with existing "on-invalid" attributes)

Reply:
Yes but how to deal with the "yaml problem" ?
As I written in the description, `null` without quotes with be interpreted as null value by YamlLoader.

Reply:
`null`should throw an exception as it would be an invalid value to me

Reply:
Ok, so the end user should write something like `decoration_on_invalid: 'null'`.
If he writes: `decoration_on_invalid: null`, this will throw an exception.
I think that the exception have to be explicit. A lot of users will do the error to write `null` instead of `'null'`

Reply:
What do you think about : `The configuration key "decoration_on_invalid" is unsupported for the service "%s" in "%s". Allowed configuration keys are "exception", "ignore" and "null". If you written "null", ensure that it is surrounded with quotes.` ?